File: content-solutions/graph-representation/graph-one-stroke-walk.py
def solution(n, trails):
    degree = [0] * n
    for a, b in trails:
        degree[a] += 1
        degree[b] += 1
    odd = sum(1 for d in degree if d % 2 == 1)
    return odd == 0 or odd == 2

File: content-solutions/graph-representation/graph-one-stroke-walk.java
class Solution {
    public boolean solution(int n, int[][] trails) {
        int[] degree = new int[n];
        for (int[] t : trails) {
            degree[t[0]]++;
            degree[t[1]]++;
        }
        int odd = 0;
        for (int d : degree) if (d % 2 == 1) odd++;
        return odd == 0 || odd == 2;
    }
}
